### PR TITLE
use choose_metric() in augment()

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
I did the `choose_metric()` call in `augment()` rather than setting `metric = NULL` in `select_best()`, as this gives us the correctly warning origin.

To close https://github.com/tidymodels/tune/issues/810

### CODE REVIEW

**Comment by EmilHvitfeldt** [comment]
``` r
library(tidymodels)
library(censored)
#> Loading required package: survival

set.seed(1)
sim_dat <- prodlim::SimSurv(500) %>%
  mutate(event_time = Surv(time, event)) %>%
  select(event_time, X1, X2) %>%
  as_tibble()

set.seed(2)
sim_rs <- vfold_cv(sim_dat)

time_points <- c(10, 1, 5, 15)

mod_spec <-
  proportional_hazards(penalty = tune(), mixture = 1) %>%
  set_engine("glmnet") %>%
  set_mode("censored regression")

grid <- tibble(penalty = 10^c(-4, -2, -1))

gctrl <- control_grid(save_pred = TRUE, save_workflow = TRUE)

mix_mtrc <- metric_set(brier_survival, brier_survival_integrated, concordance_survival)

set.seed(2193)
tune_res <-
  mod_spec %>%
  tune_grid(
    event_time ~ X1 + X2,
    resamples = sim_rs,
    grid = grid,
    metrics = mix_mtrc,
    eval_time = time_points,
    control = gctrl
  )


# `eval_time` across different functions --------------------------------

res <- select_best(tune_res)
#> Warning in select_best(tune_res): No value of `metric` was given;
#> "brier_survival" will be used.
#> Warning: 4 evaluation times are available; the first will be used (i.e. `eval_time =
#> 10`).

res <- augment(tune_res)
#> Warning in augment(tune_res): No value of `metric` was given; "brier_survival" will be used.
#> 4 evaluation times are available; the first will be used (i.e. `eval_time =
#> 10`).
```

<sup>Created on 2024-01-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: R/augment.R [hunk 1]
```diff
@@ -46,8 +46,10 @@ augment.tune_results <- function(x, parameters = NULL, ...) {
 
   # check/determine best settings
   if (is.null(parameters)) {
-    obj_fun <- .get_tune_metric_names(x)[1]
-    parameters <- select_best(x, metric = obj_fun)
+    metric_info <- choose_metric(x, metric = NULL)
+    metric <- metric_info$metric
+
```
Comment (topepo) [inline R/augment.R:51]: Maybe you are doing things in stages but I think that you have to get/set evaluation time here too: https://github.com/tidymodels/tune/blob/main/R/select_best.R#L82-L85

Can you add non-survival tests here and survival tests in extratests?

Comment (topepo) [reply R/augment.R:51]: We'll need an `eval_time` argument for these functions too.

**Comment by EmilHvitfeldt** [comment]
``` r
library(tidymodels)
library(censored)
#> Loading required package: survival

set.seed(1)
sim_dat <- prodlim::SimSurv(500) %>%
  mutate(event_time = Surv(time, event)) %>%
  select(event_time, X1, X2) %>%
  as_tibble()

set.seed(2)
sim_rs <- vfold_cv(sim_dat)

time_points <- c(10, 1, 5, 15)

mod_spec <-
  proportional_hazards(penalty = tune(), mixture = 1) %>%
  set_engine("glmnet") %>%
  set_mode("censored regression")

grid <- tibble(penalty = 10^c(-4, -2, -1))

gctrl <- control_grid(save_pred = TRUE, save_workflow = TRUE)

mix_mtrc <- metric_set(brier_survival, brier_survival_integrated, concordance_survival)

set.seed(2193)
tune_res <-
  mod_spec %>%
  tune_grid(
    event_time ~ X1 + X2,
    resamples = sim_rs,
    grid = grid,
    metrics = mix_mtrc,
    eval_time = time_points,
    control = gctrl
  )


# `eval_time` across different functions --------------------------------

res <- select_best(tune_res)
#> Warning in select_best(tune_res): No value of `metric` was given; "brier_survival" will be used.
#> Warning in show_best(x, metric = metric, n = 1, eval_time = eval_time): 
#> 4 evaluation times are available; the first will be used (i.e. `eval_time = 10`).

# similar error as `select_best()`
res <- augment(tune_res)
#> Warning in augment(tune_res): No value of `metric` was given; "brier_survival" will be used.
#> 4 evaluation times are available; the first will be used (i.e. `eval_time = 10`).

# setting `eval_time` removes part of warning
res <- augment(tune_res, eval_time = 10)
#> Warning in augment(tune_res, eval_time = 10): No value of `metric` was given; "brier_survival" will be used.

# Setting `parameters` deals with this altogether
res <- augment(tune_res, parameters = tibble(penalty = 0.1))
```

File: R/augment.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/augment.R]: `eval_time` argument added here: [`54f0d5b` (#825)](https://github.com/tidymodels/tune/pull/825/commits/54f0d5bfb754fc2aa74217aa54b448c3e744d277)

Comment (EmilHvitfeldt) [reply R/augment.R]: survival tests are done https://github.com/tidymodels/extratests/pull/178

and non-survival tests have been expanded with [`5aefeac` (#825)](https://github.com/tidymodels/tune/pull/825/commits/5aefeacebb6750ae09574a228c71823b651c319f)

**Comment by hfrick** [comment]
Apologies about the "no new eval_time arg for augment()" part getting lost in translation from https://github.com/tidymodels/tune/issues/766 to #810 -- I've updated #810 to reflect our current plans for augment() and opened #832 to address that.

**Comment by EmilHvitfeldt** [comment]
closing in favor of #832

**Pull request closed without merging**